### ISSUE TITLE
Python Crash with Piperider Execution in dg23_explorer Project

### ISSUE THREAD

**Issue by Jing-Kai** [Jing-Kai/dg23_explorer#14]
## Overview

During the execution of the piperider in conjunction with the dg23_explorer dbt project, I experienced a Python crash. The dbt project utilized versions: dbt core 1.6.3, duckdb v0.8.1 6536a77232, and piperider 0.34.0 with Python 3.11.5.

![image](https://github.com/Jing-Kai/dg23_explorer/assets/8622675/05b1e2d4-e76c-4b93-bc3e-ae8152599ac6)

## Error Details
The crash generated the following message:
```
*** Incorrect guard value: 0
Python(10445,0x1712f3000) malloc: *** set a breakpoint in malloc_error_break to debug
[1]    10445 abort      piperider run
```

## Reproduction Context:
While my teammate, Iris Chen, executed piperider run successfully, I encountered the crash. We both utilize Pipenv to manage our virtual environments, implying a consistent codebase and package set between our setups.

## System Information:
Python Process Details:
```
Process:               Python [13187]
Path:                  /opt/homebrew/*/Python.framework/Versions/3.11/Resources/Python.app/Contents/MacOS/Python
Identifier:            org.python.python
Version:               3.11.5 (3.11.5)
Code Type:             ARM-64 (Native)
Parent Process:        zsh [11401]
Responsible:           iTerm2 [1300]
User ID:               502
Date/Time:             2023-10-24 14:40:53.2031 +0800
OS Version:            macOS 13.5.2 (22G91)
Report Version:        12
Crashed Thread:        16
Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Termination Reason:    Namespace SIGNAL, Code 6 Abort trap: 6
Terminating Process:   Python [13187]
```

Installed Packages:
```
agate==1.7.1
attrs==23.1.0
Babel==2.12.1
blessed==1.20.0
boto3==1.28.53
botocore==1.31.53
certifi==2023.7.22
cffi==1.15.1
chardet==5.2.0
charset-normalizer==3.2.0
click==8.1.7
colorama==0.4.6
dbt-core==1.6.3
dbt-duckdb==1.6.0
dbt-extractor==0.4.1
dbt-semantic-interfaces==0.2.0
deepmerge==1.1.0
duckdb==0.8.1
duckdb-engine==0.9.2
fsspec==2023.9.2
hologram==0.0.16
idna==3.4
importlib-metadata==6.8.0
inquirer==3.1.3
isodate==0.6.1
Jinja2==3.1.2
jmespath==1.0.1
jsonschema==3.2.0
leather==0.3.4
Logbook==1.5.3
markdown-it-py==3.0.0
MarkupSafe==2.1.3
mashumaro==3.8.1
mdurl==0.1.2
minimal-snowplow-tracker==0.0.2
more-itertools==8.14.0
msgpack==1.0.6
networkx==3.1
packaging==23.1
parsedatetime==2.6
pathspec==0.11.2
piperider==0.34.0
pipfile==0.0.2
portalocker==2.8.2
protobuf==4.24.3
pycparser==2.21
pydantic==1.10.12
Pygments==2.16.1
pyrsistent==0.19.3
python-dateutil==2.8.2
python-editor==1.0.4
python-slugify==8.0.1
pytimeparse==1.1.8
pytz==2023.3.post1
PyYAML==6.0.1
readchar==4.0.5
requests==2.31.0
requests-toolbelt==1.0.0
rich==13.5.3
ruamel.yaml==0.17.32
ruamel.yaml.clib==0.2.7
s3fs==0.4.2
s3transfer==0.6.2
sentry-sdk==1.31.0
six==1.16.0
SQLAlchemy==2.0.21
sqlparse==0.4.4
text-unidecode==1.3
toml==0.10.2
typing_extensions==4.8.0
urllib3==1.26.16
wcwidth==0.2.6
zipp==3.17.0

```

**Comment by Jing-Kai** [comment]
@irisschen I wonder if I could know your duckdb version?

**Comment by irisschen** [comment]
My duckdb version: v0.8.1

**Comment by Jing-Kai** [comment]
submit to InfuseAI Evan Wei on slack channel 'partner-kkstream'.

**Comment by Jing-Kai** [comment]
From InfuseAI

> 從你的描述，有點像是遇到
https://github.com/duckdb/duckdb/issues/8230
解決的時間是 8/4
而你用的 duckdb 0.8.1 是  6/13 應該還沒有解決這個問題
https://github.com/duckdb/duckdb/releases/tag/v0.8.1
但是 PR 8450 有在
https://github.com/duckdb/duckdb/releases/tag/v0.9.0
目前最新的 stable release 是 v0.9.1

**Comment by Jing-Kai** [comment]
I can pass this by upgrading duckDB to v0.9.1.

**Comment by Jing-Kai** [comment]
Resolved after upgrading duckDB with PipeRider 0.34.0.
